glean-core/ios/Glean/Glean.swift

```diff
@@ -108,7 +114,46 @@ public class Glean {
 
     /// Handle background event and send appropriate pings
     func handleBackgroundEvent() {
-        // sendPings()
+        Glean.backgroundTaskId = UIApplication.shared.beginBackgroundTask(withName: "Glean Upload Task") {
+            // End the task if time expires
+            UIApplication.shared.endBackgroundTask(Glean.backgroundTaskId)
+            Glean.backgroundTaskId = UIBackgroundTaskIdentifier.invalid
+        }
+
+        self.sendPingsByName(pingNames: ["baseline", "events"])
+
+        // End task assertion
+        UIApplication.shared.endBackgroundTask(Glean.backgroundTaskId)
```

Comment:
As I understand it marking something with `beginBackgroundTask` ensures the app isn't killed right away before `endBackgroundTask` happens.
With `sendPingsByName` dispatching to the thread pool, it will return immeidately, likely before the task even started executing.
`endBackgroundTask` is then called while the Rust backend might just be reading data and writing the pings.

Shouldn't the begin/end task wrapping happen inside `sendPingsByName` inside the threadpool-launched task?

Reply:
Yes, it should...  fixing this